src/test/main-thread/object-mutation.test.ts

```diff
@@ -0,0 +1,274 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import anyTest, { TestInterface } from 'ava';
+import * as sinon from 'sinon';
+import { Strings } from '../../main-thread/strings';
+import { ObjectContext } from '../../main-thread/object-context';
+import { CommandExecutor } from '../../main-thread/commands/interface';
+import { Env } from './helpers/env';
+import { ObjectMutationProcessor } from '../../main-thread/commands/object-mutation';
+import { NodeContext } from '../../main-thread/nodes';
+import { WorkerContext } from '../../main-thread/worker';
+import { normalizeConfiguration } from '../../main-thread/configuration';
+import { TransferrableMutationType } from '../../transfer/TransferrableMutation';
+import { TransferrableObjectType } from '../../transfer/TransferrableMutation';
+
+const test = anyTest as TestInterface<{
+  sandbox: sinon.SinonSandbox;
+  strings: Strings;
+  objectContext: ObjectContext;
+  objectMutationProcessor: CommandExecutor;
+  canvasElement: HTMLCanvasElement;
+}>;
+
+test.beforeEach(t => {
+  const sandbox = sinon.createSandbox();
+  const env = new Env();
+  const { document } = env;
+  const baseElement = document.createElement('div');
+
+  const strings = new Strings();
+  const objectContext = new ObjectContext();
+
+  const ctx = {} as CanvasRenderingContext2D;
+  const canvasElement = ({ _index_: 1, getContext: (c: string) => ctx } as unknown) as HTMLCanvasElement;
+
+  const nodeContext = new NodeContext(strings, baseElement);
+  sandbox.stub(nodeContext, 'getNode').returns(canvasElement);
+  const workerContext = ({
+    getWorker() {},
+    messageToWorker() {},
+  } as unknown) as WorkerContext;
+
+  const objectMutationProcessor = ObjectMutationProcessor(
+    strings,
+    nodeContext,
+    workerContext,
+    normalizeConfiguration({
+      domURL: 'domURL',
+      authorURL: 'authorURL',
+    }),
+    objectContext,
+  );
+
+  t.context = {
+    sandbox,
+    strings,
+    objectContext,
+    objectMutationProcessor,
+    canvasElement,
+  };
+});
+
+test.afterEach(t => {
+  const { sandbox } = t.context;
+  sandbox.restore();
+});
+
+test('Method call with no arguments', t => {
+  const { sandbox, strings, objectMutationProcessor, canvasElement } = t.context;
+
+  const methodName = 'stroke';
+  const targetObject = canvasElement.getContext('2d');
+
+  if (!targetObject) {
```

Comment:
Do we need this?

Reply:
TypeScript will complain otherwise..

Reply:
More specifically, every usage below will throw `Object is possibly null`.